### ISSUE TITLE
Add support for Apple Silicon

### ISSUE THREAD

**Issue by waldekmastykarz** [dotnet/dev-proxy#529]
Based on the feedback from @tmaestrini

- [x] add support for architecture osx-arm64
- [x] update setup scripts

**Comment by waldekmastykarz** [comment]
Unfortunately, the osx-arm64 doesn't work. After starting it stops immediately. As a workaround for now, we patched the setup script to install x64 version on arm64 macs (#540) which seems to be working just fine. In parallel, we'll investigate why the arm64 version isn't working.

**Comment by jabberwik** [comment]
It works for me when I build from source. Possible to add this back to the next release?

**Comment by waldekmastykarz** [comment]
Correct, it's working when you build it on your own machine, but fails if you try to run bits built elsewhere (like on CI/CD). We suspect it has to do with code signing. We're in the process of setting it up and should have something in a few weeks. We'll share an update then. For now, you can use osx-x64 bits.

**Comment by waldekmastykarz** [comment]
To properly support it, we'd need notarization. We don't have an Apple dev account at the moment, so let's park it for now and use the x64 version instead.
